api/client/options.proto

```diff
@@ -225,4 +225,10 @@ message CommandLineOptions {
   google.protobuf.Timestamp scheduled_start = 105;
 
   reserved 38; // deprecated
+  // Provide a unique execution id. Will be reflected in the output when set.
```

Comment:
Is this a good opportunity to also describe the required format? E.g. the identifier must be something that can use used as a directory name on Linux filesystem?

Reply:
See below, `When populating this field, it is recommended...`. There will only be constraints to this value when there is a Sink involved which imposes limitations; our file based sink is picky (at least today) and only accepts unique identifiers. Maybe we can later work to remove that restriction by using a codec. For max key length I was thinking along the same lines, that might vary depending on the implementation specifics of Sinks?

Reply:
Makes sense, we don't want to expose limitations of a single sink up to the public API.

We can make sure that the individual Sink implementations will validate their inputs and return readable error messages when an incompatible `execution_id` will be specified.